test/modules/generic-mqtt-tester/src/main.rs

```diff
@@ -59,13 +58,14 @@ async fn listen_for_shutdown() -> Result<(), MessageTesterError> {
     info!("registering unix signal listeners");
     let mut interrupt =
         signal(SignalKind::interrupt()).map_err(MessageTesterError::CreateUnixSignalListener)?;
-    let mut term =
+    let mut terminate =
         signal(SignalKind::terminate()).map_err(MessageTesterError::CreateUnixSignalListener)?;
-    let interrupt_fut = interrupt.next();
-    let term_fut = term.next();
+    let interrupt = interrupt.recv();
+    let terminate = terminate.recv();
+    pin_mut!(interrupt, terminate);
```

Comment:
I am trying to read up on changes in signal, but why do we need to pin the futures now?

Reply:
Signal is not `Stream` anymore. https://docs.rs/tokio/1.5.0/tokio/signal/unix/struct.Signal.html so we need to ensure future returned by `recv` is pinned

Reply:
Ah that makes sense